src/app/attendance/actions/create-attendance.ts

```diff
@@ -17,6 +19,8 @@ export const createAttendance = async (values: AttendanceFormValues) => {
   const { error } = await supabase.from(Tables.Attendance).insert({
     ...payload,
     status: AttendanceStatus.Pending,
+    role:loginUser.role,
+    shift:loginUser.shift
```

Comment:
I'm not sure, can you please tell me why you inserting role and shift while we are getting in payload.